### PR TITLE
Fix JodaCompatibleZonedDateTime casts in Painless

### PR DESCRIPTION
**Pull request by jdconrad**
This is a temporary fix during the Joda to Java datetime transition.  This will implicitly cast a JodaCompatibleZonedDateTime to a ZonedDateTime for both def and static types.  This is necessary to insulate users from needing to know about JodaCompatibleZonedDateTime explicitly.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/Utility.java [hunk 1]
```diff
@@ -43,5 +48,15 @@ public static char StringTochar(final String value) {
         return value.charAt(0);
     }
 
+    // TODO: remove this when the transition from Joda to Java datetimes is completed
+    public static ZonedDateTime JCZDTToZonedDateTime(final JodaCompatibleZonedDateTime jczdt) {
+        return jczdt.getZonedDateTime();
+    }
+
+    // TODO: remove this when the transition from Joda to Java datetimes is completed
+    public static Temporal JCZDTToTemporal(final JodaCompatibleZonedDateTime jczdt) {
```
Comment (rjernst) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/Utility.java]: this is unused now?

**Review by rjernst — APPROVED** [review]
LGTM

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/Utility.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/Utility.java]: Good catch!  Removed.

**Comment by jdconrad** [comment]
@rjernst Thanks for the review!  Will merge as soon as CI passes.

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by jdconrad** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**